spylib/request.py

```diff
@@ -215,53 +217,116 @@ def put(self, base_url, path, payload, timeout, retry_count, **kwargs):
+        Helper method to use `self.refresh_token` to obtain a new
+        access token.
+
+        Sets `self.access_token` if the method succeeds.
+
+        Raises `AuthCredentialException` if it fails.
         """
-        base_url = os.environ.get("SPYLIB_AUTH_BASE_URL", None)
-        if not base_url:
-            raise Exception("SPYLIB_AUTH_BASE_URL must be set.")
-        if not refresh_token:
-            raise RefreshException
-        cookies = {"refresh_token": refresh_token}
-        resp = self.make_service_request(
-            base_url,
-            "/api/v1/tokens",
-            method="POST",
-            payload={},
+        if self.refresh_token is None:
+            raise AuthCredentialException(
+                "Auth refresh failed - no refresh token found"
+            )
+
+        # Make a request
+        resp = self.post(
+            AUTH_BASE_URL,
+            "api/v1/tokens",
             timeout=2,
-            cookies=cookies,
+            cookies={"refresh_token": self.refresh_token},
         )
-        if resp.status_code != 201:
-            raise RefreshException
-        access_token = resp.json().get("access_token")
-        if not access_token:
-            raise RefreshException
+
+        # Validation
+        if resp.status_code not in [201]:
+            raise AuthCredentialException(
+                "Auth refresh failed - auth returned a %d" % resp.status_code
+            )
+
+        try:
+            access_token = resp.json()["access_token"]
+        except (ValueError, KeyError):
+            raise AuthCredentialException(
+                "Auth refresh failed - failed parsing access token"
+            )
+
+        if access_token is None:
+            raise AuthCredentialException(
+                "Auth refresh failed - failed fetching access token"
+            )
+
+        # Persist the token locally
         self._set_access_token(access_token)
 
-    def login(self, uuid, api_key):
+    def _fetch_tokens_with_api_key(self):
         """
-        Login a user on auth, returns access_token and refresh_token.
+        Helper method to use `self.uuid` and `self.api_key` to log in to the auth service.
+
+        Sets `self.access_token` and `self.refresh_token` if the method succeeds.
+
+        Raises `AuthCredentialException` if it fails.
         """
-        base_url = os.environ.get("SPYLIB_AUTH_BASE_URL", None)
-        if not base_url:
-            raise Exception("SPYLIB_AUTH_BASE_URL must be set.")
-        resp = self.make_service_request(
-            base_url,
+        if not self.uuid or not self.api_key:
+            raise AuthCredentialException(
+                "Auth login failed - attempted to fetch tokens without providing API key or UUID"
+            )
+
+        # Make a request
+        resp = self.post(
+            AUTH_BASE_URL,
             "api/v1/login",
-            method="POST",
-            payload={"api_key": api_key, "uuid": uuid},
+            timeout=2,
```

Comment:
we can always remove this to simplify the call if you'd like 👍

Reply:
Good call!